src/test/shell/bazel/check_external_files.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+#
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Test check_external_files flag
+
+# Load the test setup defined in the parent directory
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${CURRENT_DIR}/../integration_test_setup.sh" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+setup_remote() {
+  WRKDIR=$(mktemp -d "${TEST_TMPDIR}/testXXXXXX")
+  cd "${WRKDIR}"
+
+  mkdir remote
+  (
+    cd remote
+    cat > BUILD <<'EOF'
+genrule(name="g", outs=["go"], cmd="echo GO > $@")
+EOF
+  )
+  tar cvf remote.tar remote
+  rm -rf remote
+
+  mkdir main
+  cd main
+  cat >> $(create_workspace_with_default_repos WORKSPACE) <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+  name="remote",
+  strip_prefix="remote",
+  urls=["file://${WRKDIR}/remote.tar"],
+)
+EOF
+}
+
+setup_local() {
+  WRKDIR=$(mktemp -d "${TEST_TMPDIR}/testXXXXXX")
+  cd "${WRKDIR}"
+
+  mkdir local_rep
+  (
+    cd local_rep
+    cat >> $(create_workspace_with_default_repos WORKSPACE) <<EOF
+EOF
+    cat > BUILD <<'EOF'
+genrule(name="g", outs=["go"], cmd="echo GO > $@")
+EOF
+  )
+
+  mkdir main
+  cd main
+  cat >> $(create_workspace_with_default_repos WORKSPACE) <<EOF
+local_repository(
+  name="local_rep",
+  path="../local_rep",
+)
+EOF
+}
+
+test_check_external_files() {
+  setup_remote
+  bazel build @remote//:g || fail "Expected build to succeed"
+
+  echo "broken file" > bazel-main/external/remote/BUILD
+  bazel build --noexperimental_check_external_files @remote//:g || fail "Expected build to succeed"
+
+  bazel build @remote//:g && fail "Expected build to fail" || true
+}
+
+test_check_all_flags_fast() {
+  setup_remote
+  bazel build --watchfs @remote//:g || fail "Expected build to succeed"
+
+  echo "broken file" > bazel-main/external/remote/BUILD
+  rm -rf bazel-bin/external/remote/go
+
+  bazel build \
+    --noexperimental_check_external_files \
+    --noexperimental_check_output_files \
+    --watchfs \
+    @remote//:g || fail "Expected build to succeed"
+
+  cat $(bazel info server_log) | \
+    grep 'Skipping scanning the filesystem for cache invalidations, yay' || fail "Should have found text"
+}
+
+test_local_repository_isnt_affected() {
```

Comment:
what happens for this and the following test if you also pass `--nocheck_experimental_output_files` and run with `--watchfs`?

Reply:
The `local_repository()` one passes but the `--override_repository` one won't notice the file changed with the current implementation.

Reply:
With your suggestion of just using that same guard they both ignore the flag now